tool/container_tool.py

```diff
@@ -25,9 +25,13 @@
 class ProjectContainerTool(BaseTool):
   """A tool for LLM agents to interact within a project's docker container."""
 
-  def __init__(self, benchmark: Benchmark, name: str = '') -> None:
+  def __init__(self,
+               benchmark: Benchmark,
+               name: str = '',
+               project_name: str = '') -> None:
     super().__init__(benchmark, name)
-    self.image_name = self._prepare_project_image()
+    project_name = project_name or benchmark.project
```

Comment:
`self.project_name`, so that `_prepare_project_image` don't have to take this parameter.
Otherwise you need to fix all usages of `_prepare_project_image`.